src/software/SfM/main_openMVG2MVE2.cpp

```diff
@@ -71,181 +72,201 @@ bool exportToMVE2Format(
+            const Mat3 & rotation = pose.rotation();
+            const Vec3 & translation = pose.translation();
+            // Focal length and principal point must be normalized (0..1)
+            const float flen = pinhole_cam->focal() / static_cast<double>(std::max(cam->w(), cam->h()));
+            out
+              << flen << " " << "0" << " " << "0" << "\n"  // Write '0' distortion values for pre-corrected images
+              << rotation(0, 0) << " " << rotation(0, 1) << " " << rotation(0, 2) << "\n"
+              << rotation(1, 0) << " " << rotation(1, 1) << " " << rotation(1, 2) << "\n"
+              << rotation(2, 0) << " " << rotation(2, 1) << " " << rotation(2, 2) << "\n"
+              << translation[0] << " " << translation[1] << " " << translation[2] << "\n";
+        }
+        else
+        {
+            // export a camera without pose & intrinsic info (export {0})
+            // see: https://github.com/simonfuhrmann/mve/blob/952a80b0be48e820b8c72de1d3df06efc3953bd3/libs/mve/bundle_io.cc#L448
+            for (int i = 0; i < 5 * 3; ++i)
+              out << "0" << (i % 3 == 2 ? "\n" : " ");
+        }
+    }
+
+    // For each feature, write to bundle: position XYZ[0-3], color RGB[0-2], all ref.view_id & ref.feature_id
+    // The following method is adapted from Simon Fuhrmann's MVE project:
+    // https://github.com/simonfuhrmann/mve/blob/e3db7bc60ce93fe51702ba77ef480e151f927c23/libs/mve/bundle_io.cc
+    for (const auto & landmarks_it : landmarks)
+    {
+      const Vec3 & exportPoint = landmarks_it.second.X;
+      out << exportPoint.x() << " " << exportPoint.y() << " " << exportPoint.z() << "\n";
+      out << 250 << " " << 100 << " " << 150 << "\n";  // Write arbitrary RGB color, see above note
+
+      // Write number of observations (features)
+      const Observations & obs = landmarks_it.second.obs;
+      out << obs.size();
+
+      for (const auto & obs_it : obs)
+      {
+        const IndexT viewId = obs_it.first;
+        const IndexT featId = obs_it.second.id_feat;
+        out << " " << viewId << " " << featId << " 0";
+      }
+      out << "\n";
+    }
+    out.close();
+
+    // Export (calibrated) views as undistorted images in parallel
+    std::cout << "Exporting views..." << std::endl;
+
+    // Create 'views' subdirectory
+    const string sOutViewsDirectory = stlplus::folder_append_separator(sOutDirectory) + "views";
+    if (!stlplus::folder_exists(sOutViewsDirectory))
+    {
+        cout << "\033[1;31mCreating directory:  " << sOutViewsDirectory << "\033[0m\n";
+        stlplus::folder_create(sOutViewsDirectory);
+    }
+
+    C_Progress_display my_progress_bar(views.size());
+
+    #pragma omp parallel for schedule(dynamic)
+    for (int i = 0; i < static_cast<int>(views.size()); ++i)
     {
-      const View * view = iter->second.get();
+      if (!bOk) continue;
+      const View * view = views.at(i).get();
```

Comment:
@seepa:

Note that if there is some non contiguous view this code can crash!
I mean views index can be sometimes not 0,1,2 but something like 10,20,30 (if we use only a subpart of an existing sfm_data)

I think that something like this would be better:
```
for (int i = 0; i < static_cast<int>(views.size()); ++i)
{
    auto view_it = views.begin();
    std::advance(view_it, i);
    const View * view = view_it->second.get();
```